telemetry/glean/metrics.yaml

```diff
@@ -229,82 +275,115 @@ ad_client:
-  user_agent:
+  request_id:
     type: string
     description: >
-      User agent of the client.  Used for error investigation only, no analytics value.
+      ID generated server-side during an ad request to enable correlating ad
+      requests and reported interactions. This will help give insight into how
+      many ad requests are not used or how many are reused.
     lifetime: application
     send_in_pings:
-      - events
+      - amp-interaction
+      - kevel-interaction
+      - request-stats
     notification_emails:
-      - gleonard@mozilla.com
+      - dmueller@mozilla.com
     bugs:
-      - https://mozilla-hub.atlassian.net/browse/AE-355
-      - https://bugzilla.mozilla.org/show_bug.cgi?id=1900898
+      - https://mozilla-hub.atlassian.net/browse/AE-577
+      - http://bugzilla.mozilla.org/TODO
     data_reviews:
-      - https://bugzilla.mozilla.org/show_bug.cgi?id=1900898
+      - http://bugzilla.mozilla.org/TODO
     expires: never
 
-  served_timestamp:
-    type: datetime
+  count_requested:
+    type: quantity
+    unit: ads
     description: >
-      Timestamp indicating when the ad was served by the Unified API to client.  This timestamp along with fetched_timestamp indicates the freshness of the ad.  If the ad is not cached on the server these two timestamps will differ by milliseconds. Should not be null.
+      Number of ads requested for a placement
     lifetime: application
     send_in_pings:
-      - events
+      - request-stats
     notification_emails:
-      - gleonard@mozilla.com
+      - dmueller@mozilla.com
     bugs:
-      - https://mozilla-hub.atlassian.net/browse/AE-355
-      - https://bugzilla.mozilla.org/show_bug.cgi?id=1900898
+      - https://mozilla-hub.atlassian.net/browse/AE-577
+      - http://bugzilla.mozilla.org/TODO
     data_reviews:
-      - https://bugzilla.mozilla.org/show_bug.cgi?id=1900898
+      - http://bugzilla.mozilla.org/TODO
     expires: never
 
-  fetched_timestamp:
-    type: datetime
+  count_returned:
+    type: quantity
+    unit: ads
     description: >
-      Timestamp indicating when the ad was fetched from the demand_source by the Unified API.  This timestamp along with served_timestamp indicates the freshness of the ad.  If the ad is not cached on the server these two timestamps will differ by milliseconds. Should not be null.
+      Number of ads returned by an upstream provider for a placement
     lifetime: application
     send_in_pings:
-      - events
+      - request-stats
     notification_emails:
-      - gleonard@mozilla.com
+      - dmueller@mozilla.com
     bugs:
-      - https://mozilla-hub.atlassian.net/browse/AE-355
-      - https://bugzilla.mozilla.org/show_bug.cgi?id=1900898
+      - https://mozilla-hub.atlassian.net/browse/AE-577
+      - http://bugzilla.mozilla.org/TODO
     data_reviews:
-      - https://bugzilla.mozilla.org/show_bug.cgi?id=1900898
+      - http://bugzilla.mozilla.org/TODO
+    expires: never
+
+  count_filtered:
```

Comment:
For reliable Addressable Inventory calculations we need to know that `count_filtered` is 100% due to dismissals and not due to some other processing error.

Reply:
Makes sense, processing errors will apply right after getting the ad responses, so if there are those errors they will reduce the count_returned from the ad provider. This one will be used for filtering applied on behalf of the user